From ecc95046a60be0fa94e574fa41a17d01a7801516 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Mar 2017 13:24:16 +0100
Subject: [PATCH] tc: flower: Fix incorrect error msg about eth type

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1422629
Upstream Status: iproute2.git commit 00697ca19ae3e

commit 00697ca19ae3e1118f2af82c3b41ac4335fe918b
Author: Roi Dayan <roid@mellanox.com>
Date:   Thu Jan 19 14:31:20 2017 +0200

    tc: flower: Fix incorrect error msg about eth type

    addattr16 may return an error about the nl msg size
    but not about incorrect eth type.

    Fixes: 488b41d020fb ("tc: flower no need to specify the ethertype")
    Signed-off-by: Roi Dayan <roid@mellanox.com>
    Reviewed-by: Paul Blakey <paulb@mellanox.com>
---
 tc/f_flower.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/tc/f_flower.c b/tc/f_flower.c
index d210b67..20f9a85 100644
--- a/tc/f_flower.c
+++ b/tc/f_flower.c
@@ -596,11 +596,8 @@ parse_done:
 		return ret;
 
 	ret = addattr16(n, MAX_MSG, TCA_FLOWER_KEY_ETH_TYPE, eth_type);
-	if (ret) {
-		fprintf(stderr, "Illegal \"eth_type\"(0x%x)\n",
-			ntohs(eth_type));
-		return -1;
-	}
+	if (ret)
+		return ret;
 
 	tail->rta_len = (((void*)n)+n->nlmsg_len) - (void*)tail;
 
-- 
1.8.3.1

